File: DesafiosPython/URI1040.py
notas = input().split(' ')
n1 = float(notas[0])
n2 = float(notas[1])
n3 = float(notas[2])
n4 = float(notas[3])
media = ((n1*2)+(n2*3)+(n3*4)+(n4*1))/10
print(f'Media: {media:.1f}')

if media >= 7.0:
    print('Aluno aprovado.')
elif media < 5:
    print('Aluno reprovado.')
elif media >= 5 and media < 7.0:
    print('Aluno em exame.')
    if media >= 5 and media < 7.0:
        exame = float(input())
        print(f'Nota do exame: {exame:.1f}')
        nota_final = (exame+media)/2
        if nota_final >= 5.0:
            print('Aluno aprovado.')
        else:
            print('Aluno reprovado.')
    print(f'Media final: {nota_final:.1f}')

File: DesafiosJava/URI1040.java
import java.text.DecimalFormat;
import java.util.Scanner;

public class URI1040 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        String[] valores = entrada.nextLine().split(" ");

        DecimalFormat decimalFormat = new DecimalFormat("#,#0.0");
        
        double nota1 = Double.parseDouble(valores[0])*2;
        double nota2 = Double.parseDouble(valores[1])*3;
        double nota3 = Double.parseDouble(valores[2])*4;
        double nota4 = Double.parseDouble(valores[3])*1;

        double media = (nota1+nota2+nota3+nota4)/10;

        System.out.println("Media: "+decimalFormat.format(media));

        if (media >= 7.0){
            
            System.out.println("Aluno aprovado.");
        }
        else if (media < 5.0){
            System.out.println("Aluno reprovado.");
        }
        else if (media >=5.0 && media <=6.9){
            System.out.println("Aluno em exame.");
            double valores2 = entrada.nextDouble();
            System.out.println("Nota do exame: "+decimalFormat.format(valores2));
            double mediaFinal = (media+valores2)/2;

            if (mediaFinal >= 5.0){
                System.out.println("Aluno aprovado.");
            }
            else{
                System.out.println("Aluno reprovado.");
            } 

            System.out.println("Media final: "+decimalFormat.format(mediaFinal));
        }

        entrada.close();         
    }
}
